addons/account/models/account_bank_statement.py

```diff
@@ -1,476 +1,245 @@
-                ('company_id', '=', company_id)
-            ], limit=1)
-        return self.env['account.journal']
-
-    @api.depends('balance_start', 'previous_statement_id')
-    def _compute_is_valid_balance_start(self):
-        for bnk in self:
-            bnk.is_valid_balance_start = (
-                bnk.currency_id.is_zero(
-                    bnk.balance_start - bnk.previous_statement_id.balance_end_real
-                )
-                if bnk.previous_statement_id
-                else True
+    def default_get(self, fields_list):
+        # EXTENDS base
+        defaults = super().default_get(fields_list)
+        # create statement on a saved statement line in the tree view
+        if self._context.get('st_line_id'):
+            st_line = self.env['account.bank.statement.line'].browse(self._context['st_line_id'])
+            defaults['balance_start'] = st_line.running_balance - st_line.amount
+            return defaults
+        # create statement from a new line in the tree view, not stored in the db yet
+        if self._context.get('st_line_date'):
+            defaults['balance_start'] = self.env['account.bank.statement.line'].search(
+                domain=[
+                    ('date', '<=', self._context['st_line_date']),
+                    ('journal_id', '=', self._context.get('st_line_journal_id')),
+                ],
+                order='internal_index desc',
+                limit=1
+            ).running_balance
+            return defaults
+        lines = None
+        # creating statements with split button
+        if self._context.get('split_line_id'):
+            current_st_line = self.env['account.bank.statement.line'].browse(self.env.context.get('split_line_id'))
+            line_before = self.env['account.bank.statement.line'].search(
+                domain=[
+                    ('internal_index', '<=', current_st_line.internal_index),
+                    ('journal_id', '=', current_st_line.journal_id.id),
+                    ('statement_id', '!=', current_st_line.statement_id.id),
+                    ('statement_id', '!=', False),
+                ],
+                order='internal_index desc',
+                limit=1,
+            )
+            lines = self.env['account.bank.statement.line'].search(
+                domain=[
+                    ('internal_index', '<=', current_st_line.internal_index),
+                    ('internal_index', '>', line_before.internal_index or ''),
+                    ('journal_id', '=', current_st_line.journal_id.id),
+                ],
+                order='internal_index desc',
+            )
+        # if it is called from the action menu, we can have both start and end balances and we filter out
+        # completed statements, because it is probably due to a mistake from the user
+        elif self._context.get('active_model') == 'account.bank.statement.line' and self._context.get('active_ids'):
```

Comment:
`'account.bank.statement.line'` and `'active_ids'` are filled automatically in kanban, so in both cases they are available.
I could use something like `is_split` instead of `split_line_id` and use active_id, but I think this way is more explicit and makes debugging easier.